File: algorithms/Masking Personal Information/Masking Personal Information.java
class Solution {
    public String maskPII(String s) {
        StringBuilder sb  = new StringBuilder();
		         //email handeling
        if((s.charAt(0) >= 97 && s.charAt(0) <= 122) || (s.charAt(0) >= 65 && s.charAt(0) <= 90)){

            s = s.toLowerCase();
           int indexofAt = s.indexOf('@');
           String firstName = s.substring(0, indexofAt);
           sb.append(firstName.charAt(0)).append("*****").append(firstName.charAt(firstName.length()-1));
           sb.append(s.substring(indexofAt,s.length()));
        }
        //phone number handeling
        else{
           int digits = 0;
           for(int i = 0 ; i < s.length(); i++){
               if(Character.isDigit(s.charAt(i))){
                   digits++;
               }
           }
           if(digits > 10){
               sb.append('+');
           }
           while(digits > 10){
               sb.append('*');
               digits--;
           }
           if(sb.toString().isEmpty() == false){
               sb.append('-');
           }
           sb.append("***").append('-').append("***-");
           StringBuilder last4 = new StringBuilder();
           int count = 0;
           for(int i  = s.length()-1; i >=0; --i){
               if(count == 4){
                   break;
               }
               if(Character.isDigit(s.charAt(i))){
                   last4.append(s.charAt(i));
                   count++;
               }
           }
           sb.append(last4.reverse());
        }

        return sb.toString();
    }
}

File: scripts/algorithms/M/Masking Personal Information/Masking Personal Information.cpp
class Solution {
public:
  string email(string &s, int id){
    string t = string(1, s[0] | ' ') + "*****" + string(1, s[id-1] | ' ');
  
    for(int i = id + 1; i != s.size(); i++) 
      if(s[i] != '.') s[i] |= ' ';
    
    return t + s.substr(id, s.size());
  }
  
  string phone(string &s){
    string table[] = { "" , "+*-", "+**-", "+***-" }, t;
    
    for(auto &ch: s)
      if(ch >= '0' && ch <= '9') t.push_back(ch);
   
    return table[t.size() - 10] + "***-***-" + t.substr(t.size() - 4, 4);
  }
  
  string maskPII(string s) {
    int id = s.find('@'); 
    return id == -1 ? phone(s) : email(s, id);
  }
};